### DISCUSSION TITLE
Supposed way to add custom link preload tags in v14?

### DISCUSSION THREAD

**Discussion by janrembold** [vercel/next.js#57877] [Help] [answered]
### Summary

Hi,

I'm searching for a way to add custom preload tags to my statically exported website. I'm using the latest version of NextJS and want to achieve something like this, probably with Metadata:

```
<Head>
                <link
                    rel="preload"
                    href="/images/poster.webp"
                    as="image"
                    fetchPriority="high"
                />
                <link
                    rel="preload"
                    href="/videos/demo.webm"
                    as="video"
                    fetchPriority="high"
                />
</Head>
```

but the only way that is close but not equal is using the `icons` prop. The `as` prop is not supported here.

```
export const metadata: Metadata = {
    title: 'Metadata Page Title',
    icons: {
        other: [
            {
                rel: 'preload',
                url: '/videos/poster.webp',
                fetchPriority: 'high',
            },
            {
                rel: 'preload',
                url: '/videos/demo.webm',
                fetchPriority: 'high',
            },
        ],
    },
};
```

As preloading is a crucial part of website optimization, I'm confused as to why this doesn't seem to be better supported by Next. Or, am I missing something here?

Really appreciate any help!
Thanks!
Best, Jan

**Comment by icyJoseph** [comment]
Hi,

- https://nextjs.org/docs/app/api-reference/functions/generate-metadata#resource-hints

You have to use the new ReactDOM methods.

**Reply by janrembold** [reply]
Thanks for the quick reply! I really missed that part. 

But adding this piece of code doesn't add the preload tags. 
I tried to put it in the file /app/preload-resources.tsx as mentioned in the docs, and also exported it from the client component where I would actually expect this piece of code. Anything else I'm missing here?

Btw, I'm using latest packages: next@14.0.1 and react-dom@18.2.0

```
'use client ';

import ReactDOM from 'react-dom';

export function PreloadResources() {
    ReactDOM.preload('/videos/poster.webp', {
        as: 'image',
        fetchPriority: 'high',
    });
    ReactDOM.preload('/videos/demo.webm', {
        as: 'video',
        fetchPriority: 'high',
    });

    return null;
}
```

**Reply by icyJoseph** [reply, marked as answer]
Aha, I don't think the preload-resources is a special file necessarily, it's just an example.

For example you can import `PreloadResources` into `app/layout.tsx` and render it somewhere in that tree.

**Reply by janrembold** [reply]
Now i was really at a loss :D Of course, simply using `PreloadResources` as a jsx/tsx component and rendering it inside the desired client component did the trick!!! Thanks a ton @icyJoseph

**Reply by khanakia** [reply]
Getting error Type '"image"' is not assignable to type 'PreloadAs'.ts(2322) 

I can see the code only ` type PreloadAs = 'font' | 'script' | 'style';` these are allowed

**Reply by erfanansari** [reply]
Unfortunately this method will overwrite the metatags provided in the layout file, so not the best way around

**Comment by tranvanhieu01012002** [comment]
For me:
I use this way in ```page.jsx```

```
import { preload } from "react-dom";

export default async function Home({ params }) {
  const background = "background url";
  preload(background, { as: "image", fetchPriority: "hight" });

  return <>{/* Your content */}</>;
}


```

**Reply by HakanSungur** [reply]
I use preload with image tag all is right but ı use in as='style' doesnt work for me.
